query02.sql

```diff
@@ -0,0 +1,36 @@
+/**
+2. Which bus stop has the smallest population within 800 meters?
+
+  **The queries to #1 & #2 should generate relations with a single row, with the following structure:**
+
+  ```sql
+  (
+      stop_name text, -- The name of the station
+      estimated_pop_800m integer, -- The population within 800 meters
+      the_geom geometry(Point, 4326) -- The geometry of the bus stop
+  )
+  ```
+**/
+with bus_stop_geom as (		
+select stop_id, stop_name, ST_SetSRID(ST_Point(stop_lon, stop_lat),4326) As geometry
+	FfromROM septa_bus_stops
+),
+bus_stop_block_group as (
+	select b.stop_id, b.stop_name, bg.geoid10 as geoid
+	from bus_stop_geom as b
+	join census_block_groups as bg
+	on ST_Intersects(
+		ST_Buffer(ST_Transform(b.geometry, 32129),800),
```

Comment:
Because of this line, the query completes in a reasonable time for me _only if_ I add an index such as:
```sql
drop index if exists  septa_bus_stops__geom_32129__buffer;
create index septa_bus_stops__geom_32129__buffer
	on septa_bus_stops
	using GiST (ST_Buffer(ST_Transform(ST_SetSRID(ST_Point(stop_lon, stop_lat), 4326), 32129), 800));
```

Instead of using `ST_Intersects(ST_Buffer(...`, you could use `ST_DWithin(...`, and then could at least get rid of the `ST_Buffer` in the above index and still complete in a reasonable time.